istioctl/cmd/workload.go

```diff
@@ -589,3 +599,29 @@ func mapToString(m map[string]string) string {
 	sort.Strings(lines)
 	return strings.Join(lines, "\n") + "\n"
 }
+
+// extractClusterIDFromInjectionConfig can extract clusterID from injection configmap
+func extractClusterIDFromInjectionConfig(kubeClient kube.ExtendedClient) string {
+	injectionConfigMap := "istio-sidecar-injector"
+	// Case with multiple control planes
+	revision := kubeClient.Revision()
+	if revision != "" && revision != "default" {
+		injectionConfigMap = fmt.Sprintf("%s-%s", injectionConfigMap, revision)
+	}
+	istioInjectionCM, err := kubeClient.CoreV1().ConfigMaps(istioNamespace).Get(context.Background(), injectionConfigMap, metav1.GetOptions{})
+	// retuen empty if the requested configmap does not exist in the given namespace
+	if err != nil {
+		log.Warnf("configmap %s was not found in namespace %s: %v", istioInjectionCM, istioNamespace, err)
```

Comment:
Instead of logging, which looks terrible in _istioctl_, can we write to stderr like the other commands when they encounter problems?

Reply:
sure, done!